ExperimentDatabase.py

```diff
@@ -51,17 +47,30 @@ def setup_experiment(self, name, species, uses_rfid, num_animals, num_groups, ca
                             (name, species, uses_rfid, num_animals, num_groups, cage_max))
         self._conn.commit()
 
-    def setup_groups(self, group_names, animals_per_group):
+    def setup_groups(self, group_names):
         '''Adds the groups to the database.
             arg1 (list): a list of all the group names
-            arg2 (int): a number representing the number of animals per group
         '''
         for group in group_names:
-            self._c.execute(''' INSERT INTO groups (name, num_animals) 
-                                VALUES (?, ?)''',
-                                (group, animals_per_group))
+            self._c.execute(''' INSERT INTO groups (name, num_animals, full) 
+                                VALUES (?, ?, ?)''',
+                                (group, 0, 0))
             self._conn.commit()
 
+    def setup_cages(self, num_animals, num_groups, cage_max):
+        cages_per_group = (int(num_animals)/int(num_groups)) / int(cage_max)
+        
+        group = 1
+        cage = 1
+        for x in range(1, int(num_groups)+1):
+            for cage in range(int(cages_per_group)):
+                self._c.execute(''' INSERT INTO cages (group_id, num_animals, full) 
+                                    VALUES (?, ?, ?)''',
+                                    (x, 0, 0))
+                self._conn.commit()
```

Comment:
Committing to the Database for each run of for loop is not ideal. Let's discuss how to deal with this. !

Reply:
I think we need to insert everytime because we need to create that many cages.

Reply:
I will be working on this for my next PR